index.js

```diff
@@ -2,16 +2,21 @@ export default function filterObject(object, predicate) {
 	const result = {};
 	const isArray = Array.isArray(predicate);
 
-	for (const [key, value] of Object.entries(object)) {
-		if (isArray ? predicate.includes(key) : predicate(key, value, object)) {
-			Object.defineProperty(result, key, {
-				value,
-				writable: true,
-				enumerable: true,
-				configurable: true,
-			});
+	for (const key in object) {
```

Comment:
You need to add a short comment on why this is not using `Object.entries` or `Object.keys` to prevent this being reverted in the future.

Reply:
Done :+1: